contracts/libraries/LibUtils.sol

```diff
@@ -302,33 +236,53 @@ library LibUtils {
         revertIfArchProfileIs(false, archaeologist);
     }
 
-    /// @notice Gets an archaeologist given the sarcophagus identifier and the
-    /// archaeologist's address.
-    /// @param sarcoId the identifier of the sarcophagus
-    /// @param archaeologist the address of the archaeologist
-    /// @return The archaeologist
-    function getArchaeologist(bytes32 sarcoId, address archaeologist)
+    /// @notice Checks if digging fee the embalmer has supplied for
+    /// an archaeologist is greater than or equal to the arch's min digging fee
+    /// on their profile
+    ///
+    /// @param diggingFee the digging fee supplied by the embalmer
+    /// @param archaeologist the archaeologist to check minimum digging fee of
+    function revertIfDiggingFeeTooLow(uint256 diggingFee, address archaeologist)
         internal
         view
-        returns (LibTypes.ArchaeologistStorage memory)
     {
         AppStorage storage s = LibAppStorage.getAppStorage();
 
-        return s.sarcophagusArchaeologists[sarcoId][archaeologist];
+        if (diggingFee < s.archaeologistProfiles[archaeologist].minimumDiggingFee) {
+            revert LibErrors.DiggingFeeTooLow(diggingFee, archaeologist);
+        }
     }
 
-    /// @notice Checks if a sarcophagus has been finalized by checking if it
-    /// contains any arweaveTxIds.
+    /// @notice Checks if the resurrection time supplied for the sarcophagus
+    /// is within the window that an archaeologist will accept
+    ///
+    /// @param resurrectionTime the resurrectionTime supplied for the sarcophagus
+    /// @param archaeologist the archaeologist to check the max rewrap interval of
+    function revertIfResurrectionTimeTooFarInFuture(uint256 resurrectionTime, address archaeologist)
+        internal
+        view
+    {
+        AppStorage storage s = LibAppStorage.getAppStorage();
+        uint256 maxResurrectionTime = block.timestamp + s.archaeologistProfiles[archaeologist].maximumRewrapInterval;
```

Comment:
Love the logical distinction we can now make between (max) resurrection *time* and (max) *rewrap interval* because of your simple renames 🔥